js/stock-exchange.js

```diff
@@ -0,0 +1,153 @@
+const STOCKS_BASE_URL =
+    "https://stock-exchange-dot-full-stack-course-services.ew.r.appspot.com";
+const STOCKS_SEARCH_LIMIT = 10;
+const SYMBOL_PREFIX = "$";
+const API_PROFILE_COMPANY_LIMIT = 3;
+
+let search = document.getElementById("btnStockSearch");
+let inputStockSearch = document.getElementById("inputStockSearch");
+var lastSymbol = "";
+
+inputStockSearch.addEventListener("keypress", function (event) {
+    if (event.key === "Enter") {
+        event.preventDefault();
+
+        // Enter key on input works as button click, makes Fibonacci calculation
+        document.getElementById("btnStockSearch").click();
+    }
+});
+
+search.addEventListener("click", function () {
+
+    removeStockProfileInfo();
+
+    let searchText = document.getElementById("inputStockSearch").value;
+
+    if (searchText !== "") {
+        let searchURL = `${STOCKS_BASE_URL}/api/v3/search?query=${searchText}&amp;limit=${STOCKS_SEARCH_LIMIT}&amp;exchange=NASDAQ`;
+        let stockList = document.getElementById("companyProfile");
+
+        if (stockList) {
+            stockList.replaceChildren();
+            document.getElementById("stockPlot").style.display = "none";
+        }
+
+        showLoading();
+
+        fetch(searchURL)
+            .then((response) => response.json())
+            .then(function (data) {
+                hideLoading();
+                if (typeof data === "object") {
+                    let cards = [];
+
+                    cards = createStockCards(data);
+                    let stockList = document.getElementById("stockList");
+
+                    stockList.replaceChildren(...cards);
+                }
+            });
+    }
+});
+
+function createStockCards(data) {
+    let cards = [];
+    for (element of data) {
+        let card = document.createElement("div");
+        card.className = "div__card-stock card shadow cursor-pointer";
+
+        let cardBody = document.createElement("div");
+        cardBody.className = "card-body";
+
+        let titleURL = document.createElement("a");
+        titleURL.className = "card-link";
+
+        titleURL.href = "/company.html?symbol=" + element.symbol;
```

Comment:
add "." before the slash ("/") in the path in order to make sure the url will work on all platforms. like this:
titleURL.href = "./company.html?symbol=" + element.symbol;